src/components/CheckBoxList/check-box-list.tsx

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { FunctionComponent, useCallback, useState } from 'react';
+import { Box, Divider, IconButton, List, ListItem } from '@mui/material';
+import DragIndicatorIcon from '@mui/icons-material/DragIndicator';
+import { Draggable } from 'react-beautiful-dnd';
+import CheckBoxItem from './check-box-item';
+
+interface CheckboxListProps {
+    itemRenderer?: (params: {
+        item: any;
+        index: number;
+        checked: boolean;
+        toggleSelection: (id: any) => void;
+    }) => React.ReactNode;
+    values: any[];
```

Comment:
Maybe we should accept to define a type for `values` which is an array of object defining `id` and `label`

Reply:
Some time we use uuid instead of id like for network modifications